From 1591ca3dbeb5698e89c5b68a52d1356e59f4372c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 29 Jun 2016 14:45:30 +0300
Subject: [PATCH 0312/1345] fix: net: mvpp2x: gop: Disable IDLE for link up
 indication for XLG

commit  4b1b6a1fd2484b11aa31b5b268d5921b03f5c6b5 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

-disable IDLE for link up indication for XLG and bring up port
 without IDLE packets

Change-Id: I795363424845122af0826f8805079b10ffccbaee
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30846
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c |    2 ++
 .../ethernet/marvell/mvpp2x/mv_gop110_hw_type.h    |    4 ++++
 2 files changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
index aef6281..69ff770 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
@@ -1850,6 +1850,8 @@ int mv_gop110_xlg_mac_mode_cfg(struct gop_hw *gop, int mac_num,
 					MV_XLG_MAC_CTRL4_FORWARD_PFC_EN_OFFS);
 	U32_SET_FIELD(val, MV_XLG_MAC_CTRL4_FORWARD_802_3X_FC_EN_MASK, 1 <<
 				MV_XLG_MAC_CTRL4_FORWARD_802_3X_FC_EN_OFFS);
+	U32_SET_FIELD(val, MV_XLG_MAC_CTRL4_EN_IDLE_CHECK_FOR_LINK_MASK, 0 <<
+				MV_XLG_MAC_CTRL4_EN_IDLE_CHECK_FOR_LINK_OFFS);
 	mv_gop110_xlg_mac_write(gop, mac_num, reg_addr, val);
 
 	/* Jumbo frame support - 0x1400*2= 0x2800 bytes */
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
index 40c3994..90b4a7a 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
@@ -1152,6 +1152,10 @@
 #define MV_XLG_MAC_CTRL4_MAC_MODE_DMA_1G_MASK    \
 		(0x00000001 << MV_XLG_MAC_CTRL4_MAC_MODE_DMA_1G_OFFS)
 
+#define MV_XLG_MAC_CTRL4_EN_IDLE_CHECK_FOR_LINK_OFFS		14
+#define MV_XLG_MAC_CTRL4_EN_IDLE_CHECK_FOR_LINK_MASK    \
+		(0x00000001 << MV_XLG_MAC_CTRL4_EN_IDLE_CHECK_FOR_LINK_OFFS)
+
 /* Port Mac Control5 */
 #define MV_XLG_PORT_MAC_CTRL5_REG			(0x0088)
 #define MV_XLG_MAC_CTRL5_TXIPGLENGTH_OFFS		0
-- 
1.7.9.5

